### ISSUE TITLE
ActiveRecord errors while handling error without Bundler

### ISSUE THREAD

**Issue by JoshCheek** [rails/rails#32830]
### Synopsis

When running ActiveRecord, it tries to require the adapter. When requiring the adapter fails, it catches the error message and tries to reraise it with a helpful error explaining that you need to install the gem. When Bundler is loaded, it calls the methods that Bundler monkey patches, which raises a `Gem::LoadError`, and everything works. However, without Bundler, the non-monkey patched version is called, which raises a `Gem::MissingSpecError`.  These aren't compatible because they are initialized differently:

```ruby
Gem::LoadError.instance_method(:initialize).parameters
# => [[:rest]]

Gem::MissingSpecError.instance_method(:initialize).parameters
# => [[:req, :name], [:req, :requirement]]
```

So, when ActiveRecord tries to reraise it ([here](https://github.com/rails/rails/blob/46b46e85c72f78e869a1b0038040849911c7e2c5/activerecord/lib/active_record/connection_adapters/connection_specification.rb#L197)), it explodes. 

I've also reported this to Bundler [here](https://github.com/bundler/bundler/issues/6512). There's a bit more analysis there, I'd reproduce it here, but my cat is interfering with my typing.

### Steps to reproduce

Invoke `bash run.sh` from [this](https://gist.github.com/JoshCheek/a0d3af7d6dce0450fbfcda3f1aa2cb19) gist.

### Expected behavior

An error with the message  "Error loading the 'sqlite3' Active Record adapter. Missing a gem it depends on? sqlite3 is not part of the bundle. Add it to your Gemfile."

### Actual behavior

An error with the message "wrong number of arguments (given 1, expected 2)"

### System configuration

**ActiveRecord version**: 5.2.0
**Ruby version**: 2.5.0
**Rubygems version**: 2.7.6
**Rubygems version**: 1.16.1

**Comment by JoshCheek** [comment]
* I can still reproduce this error on master.
* @y-yagi's PR, https://github.com/rails/rails/pull/32850 fixes it, but IDK, they forgot to merge it or something.
* I've updated [the gist](https://gist.github.com/JoshCheek/a0d3af7d6dce0450fbfcda3f1aa2cb19) to show that it is still broken on master but is now fixed on [y-yagi:fixes_32830](https://github.com/y-yagi/rails/tree/fixes_32830)

**Comment by JoshCheek** [comment]
I don't seem to have a reopen button.

**Comment by JoshCheek** [comment]
❤️s for @rafaelfranca!!

**Comment by y-yagi** [comment]
Fixed by #34557.
